client/contract.go

```diff
@@ -0,0 +1,70 @@
+package client
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/ava-labs/avalanchego/cache"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+const (
+	contractCacheSize = 1024
+)
+
+// ContractClient is a client for the calling contract information
+type ContractClient struct {
+	ethClient *ethclient.Client
+	cache     *cache.LRU
+}
+
+// NewContractClient returns a new ContractInfo client
+func NewContractClient(endpoint string) (*ContractClient, error) {
+	endpoint = strings.TrimSuffix(endpoint, "/")
+
+	c, err := ethclient.Dial(fmt.Sprintf("%s%s", endpoint, prefixEth))
+	if err != nil {
+		return nil, err
+	}
+
+	cache := &cache.LRU{Size: contractCacheSize}
+
+	return &ContractClient{
+		ethClient: c,
+		cache:     cache,
+	}, nil
+}
+
+// ContractInfo returns the ContractInfo for a specific address
+func (c *ContractClient) ContractInfo(contractAddress common.Address, isErc20 bool) (*ContractInfo, error) {
+	cachedInfo, isCached := c.cache.Get(contractAddress)
+
+	if isCached {
+		castCachedInfo := cachedInfo.(ContractInfo)
```

Comment:
You'll need to cast this as `(*ContractInfo)` after making the pointer change below